app/Libs/ProductApiDriver.php

```diff
@@ -0,0 +1,150 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: macintosh
+ * Date: 8/1/16
+ * Time: 1:20 PM
+ */
+
+namespace App\Libs;
+
+
+class ProductApiDriver
+{
+    public $apiUrl;
+    public $httpClient;
+
+    public function __construct() {
+        $this->apiUrl = env('CDS_PRODUCT_API_URL', '');
+        $this->httpClient = new \GuzzleHttp\Client([
+            'base_uri' => $this->apiUrl
+        ]);
+    }
+
+    private function getResponse($response) {
+        $content = $response->getBody()->getContents();
+        $resJson = json_decode($content);
+
+        return $resJson;
+    }
+
+    public function listAll($otherData, $page = 1) {
+        extract($otherData);
+        $orderQs = '';
+        if(isset($orderBy) && !empty($orderBy)) {
+            $orderQs = '&orderby='.$orderBy;
+            $orderQs .= '&order='.(isset($order) ? $order : 'desc');
+        };
+        $response = $this->httpClient->get('/api/product/list?page='.$page.$orderQs);
+
+        return $this->getResponse($response);
+    }
+
+    public function create($data) {
+        $response = $this->httpClient->post('/api/product/create', [
+            'form_params' => [
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "form_params" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.relipa.blog/coding_rules#rule_key=php%3AS1192)